src/Line.cpp

```diff
@@ -6,65 +6,114 @@
 #include "../include/roboteam_utils/LineSegment.h"
 
 namespace rtt {
+Line::Line(const LineSegment &other) {
+    start=other.start;
+    end=other.end;
+}
+double Line::length() const {
+    return (end - start).length();
+}
+
+double Line::length2() const {
+    return (end - start).length2();
+}
+
+double Line::slope() const {
+    return (end.y - start.y)/(end.x - start.x);
+}
+
+bool Line::isVertical() const {
+    return (end.x == start.x) && (end.y != start.y);
+}
+
+Vector2 Line::direction() const {
+    return Vector2(end - start);
+}
+
+double Line::intercept() const {
+    return start.y - this->slope()*start.x;
+}
+
+std::pair<double, double> Line::coefficients() const {
+    return {slope(), intercept()};
+}
+
+bool Line::isPoint() const {
+    return start == end;
+}
 
-    double Line::distanceToLine(const Vector2 &point) const {
-        return (this->project(point) - point).length();
+bool Line::isParallel(const Line &line) const {
+    // check if line is vertical, otherwise check the slope
+    if (this->isVertical() || line.isVertical()) {
+        return this->isVertical() && line.isVertical();
     }
+    return this->slope() == line.slope();
+}
+bool Line::isParallel(const LineSegment &line) const {
+    // check if line is vertical, otherwise check the slope
+    if (this->isVertical() || line.isVertical()) {
+        return this->isVertical() && line.isVertical();
+    }
+    return this->slope() == line.slope();
```

Comment:
hmmm, wouldn't slope be 0 if isVertical, meaning that this->slope() == line.slope() even if they're vertical? What's the other check for?

Also, if the slope is a double or whatever then floating point in accuracy could be fatal.

Reply:
This is why we need tests

Reply:
if isVertical this->slope would be -inf or inf. There's no good way to combat this since lines may be defined by any two points abritarily close to each other making the error upon divison that much worse. Most importantly, we have never used this method, so I don't see why we should worry too much about it now. The tests written for the most important intuitive cases work and this would work just fine on the protobuf data too.

Reply:
(As in look at LineTests. There's literally almost more lines written for testing this class then there is for the class itself)

Reply:
yeah im not good at math, i thought it would be 0, but that's horizontal